backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -61,47 +50,53 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     }
     String username = email.split("@")[0];
 
-    // Create and sort queries by time
-    // TODO: eventually sort by startTimeAvailable
-    Query query = new Query(KEY_MATCH).addSort(PROPERTY_TIMESTAMP, SortDirection.DESCENDING);
-
+    // Get DatastoreService and instiate Match and Participant Datastores
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    MatchDatastore matchDatastore = new MatchDatastore(datastore);
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
 
-    // Convert list of entities to list of matches
-    List<Match> matches = new ArrayList<Match>();
-    for (Entity entity : results.asIterable()) {
-      long id = (long) entity.getKey().getId();
-      Participant firstParticipant = (Participant) entity.getProperty(PROPERTY_FIRSTPARTICIPANT);
-      Participant secondParticipant = (Participant) entity.getProperty(PROPERTY_SECONDPARTICIPANT);
-      int duration = (int) entity.getProperty(PROPERTY_DURATION);
-      long timestamp = (long) entity.getProperty(PROPERTY_TIMESTAMP);
-      Match match = new Match(id, firstParticipant, secondParticipant, duration, timestamp);
-      matches.add(match);
+    // Find participant's current match, if exists and not returned yet
+    Participant participant = participantDatastore.getParticipantFromUsername(username);
+    if (participant == null) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "Participant with username " + username + "is invalid.");
```

Comment:
Changed wording of the comment. This case really should never happen since all participants are added to the datastore in this PR. However, I'm changing this in another PR where I'll have Users vs Participants, in which case your suggestion would be more crucial.